### DISCUSSION TITLE
Using JSR with Bun

### DISCUSSION THREAD

**Discussion by jordanbtucker** [oven-sh/bun#11897] [General]
The [JavaScript Registry (JSR)](https://jsr.io/) is a modern package registry for JavaScript and TypeScript. JSR works with many runtimes (Node.js, Deno, browsers, and more) and is backwards compatible with npm.

You can also use it with Bun.

Let's say you want to use [`@std/encoding`](https://jsr.io/@std/encoding) from the Deno Standard Library hosted on JSR.

Run the following command in your project directory:

```
bunx jsr add --bun @std/encoding@^0.224.3 # including the version range is recommended
```

Now you can use imports like this, complete with typings and documentation.

```ts
import { decodeBase64Url, encodeBase64Url } from "@std/encoding/base64url";
```

---

For the nitty gritty details, this works because `jsr` adds a [scoped registry](https://bun.sh/docs/install/registries) to `bunfig.toml`, and adds an [alias](https://bun.sh/guides/install/npm-alias) to a scoped package as a dependency in `package.json` that looks like this:

```
"@std/encoding": "npm:@jsr/std__encoding@^0.224.3"
```

The following command is equivalent to running `bunx jsr add --bun @std/encoding@^0.224.3`:

```
bun add @std/encoding@npm:@jsr/std__encoding@^0.224.3
```

---

Note: The `--bun` argument is only required if you do not yet have a `bun.lockb` file or if any other package manager's lock files exist (`package-lock.json`, `pnpm-lock.yaml`, `yarn.lock`, etc.). In other words, running `jsr add` is equivalent to running `jsr add --bun` if a `bun.lockb` file exists and no other lock files exist.

Note: `bun add` currently doesn't resolve version numbers when using [aliases](https://bun.sh/guides/install/npm-alias) like `npm` does. For example, running `bun add @std/encoding@npm:@jsr/std__encoding` will result in `"@std/encoding": "npm:@jsr/std__encoding"` being added as a dependency, while running `npm install @std/encoding@npm:@jsr/std__encoding` will add `"@std/encoding": "npm:@jsr/std__encoding@^0.224.3"` (note the version automatically being resolved). For this reason, `jsr add --bun` will not resolve version numbers either, so it's recommended to add the version or [range](https://github.com/npm/node-semver?tab=readme-ov-file#advanced-range-syntax) manually.

**Comment by MarArMar** [comment]
❯ bunx jsr add @ryoppippi/unplugin-typia
error: Script not found "jsr"
❯ bun upgrade
Congrats! You're already on the latest version of Bun (which is v1.1.18)

**Reply by MarArMar** [reply]
Created an issue here https://github.com/oven-sh/bun/issues/12465
